### PR TITLE
[docs] updated contribution.md with create local branch & directory structure

### LINKED ISSUE #2131: docs: update contributing guide
### Description
As a developer, I want to update the contributing guide with the following information
1. Update pull request process and add more details. Can take reference from [here](https://github.com/hyperledger/cactus/blob/main/CONTRIBUTING.md#create-local-branch)
2. Addition of commit lint workflow to streamline commit messages. Can take reference from [here](https://github.com/hyperledger/cactus/blob/main/.github/workflows/commitlint-pull-request.yml)

### Acceptance Criteria
1. Addition of commit lint workflow mandatory on all PR to dev and main branches
2. Updation of contributing guide.

### PR DESCRIPTION
**Pull request by maheshkasabe**
**Change log**
Rename shared roles as shared_role_name to just role_name
Update Contribution.md with create local branch & directory structure

Fixes #2131 #2129

### CODE REVIEW

**Review by adityajoshi12 — COMMENTED** [review]
Please refer https://github.com/hyperledger/bevel/blob/main/CONTRIBUTING.md

**Comment by suvajit-sarkar** [comment]
kindly sign your commits, DCO lint failing

**Comment by adityajoshi12** [comment]
Please refer contribution guide https://github.com/hyperledger/bevel/blob/main/CONTRIBUTING.md
Also, PR should be for develop branch

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,6 +51,84 @@ For source code integrity , Hyperledger Bevel GitHub pull requests are accepted
...
+
+7. Repeat step 3 to 6 when you need to prepare posting new pull request.
+
+NOTE: Once you submitted pull request to Bevel repository, step 6 is not necessary when you made further changes with `git commit --amend` since your amends will be sent automatically.
+
+NOTE: You can refer original tutorial ['GitHub Standard Fork & Pull Request Workflow'](https://gist.github.com/Chaser324/ce0505fbed06b947d962)
+
+### Directory structure
+
+Whenever you begin to use your codes on Hyperledger Bevel, you should follow the directory strecture on Hyperledger Bevel.
```
Comment (jagpreetsinghsasan) [inline CONTRIBUTING.md:105]: `strecture` typo

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
There are certain points I would like to highlight here
1. Please make sure to squash merge commits into 1
2. Please add the PR description (the PR description template auto populates, which I think you have removed)

Points 1, 2 and some more improvements and mentions are required even for the contributing guide issue #2131 , but as this PR only focuses on adding the docs for creating local branches and explaining the basic folder structure, its fine.

**Pull request closed without merging**